### PR TITLE
Resource#whenWithinRenderOutsideViewport resolve when within viewport

### PR DESCRIPTION
**Pull request by keithwrightbos**
Resource#whenWithinRenderOutsideViewport never resolves if resource is within viewport but user never scrolls.  Fix is to explicitly resolve the promise by having Resource#isInViewport call Resource#resolveRenderOutsideViewport_ if true.

### CODE REVIEW

File: src/service/resource.js [hunk 1]
```diff
@@ -901,7 +901,11 @@ export class Resource {
    * @return {boolean}
    */
   isInViewport() {
-    return this.element.isInViewport();
+    const isInViewport = this.element.isInViewport();
+    if (isInViewport) {
```
Comment (bradfrizzell) [inline src/service/resource.js]: return this.element.isInViewport() ? this.resolveRenderOutsideViewport_() || true : false;

Comment (bradfrizzell) [reply src/service/resource.js]: image

Comment (bradfrizzell) [reply src/service/resource.js]: (don't actually do this)

Comment (jridgewell) [reply src/service/resource.js]: Let's not golf.

Comment (bradfrizzell) [reply src/service/resource.js]: agreed :D

**Pull request merged**